addons/gamification/models/badge.py

```diff
@@ -139,13 +139,15 @@ def _get_owners_info(self):
             the total number of users this badge was granted to
         """
         self.env.cr.execute("""
-            SELECT badge_id, count(user_id) as stat_count,
-                count(distinct(user_id)) as stat_count_distinct,
-                array_agg(distinct(user_id)) as unique_owner_ids
-            FROM gamification_badge_user
-            WHERE badge_id in %s
-            GROUP BY badge_id
-            """, [tuple(self.ids)])
+            SELECT bu.badge_id, count(bu.user_id) as stat_count,
+                count(distinct(bu.user_id)) as stat_count_distinct,
+                array_agg(distinct(bu.user_id)) as unique_owner_ids
+            FROM gamification_badge_user bu
+            JOIN res_company_users_rel r ON r.user_id = bu.user_id
+            WHERE bu.badge_id in %s
+              AND r.cid = %s
+            GROUP BY bu.badge_id
+            """, [tuple(self.ids), self.env.company.id])
```

Comment:
You should limit to *all* accessibles companies
```suggestion
              AND r.cid IN %s
            GROUP BY bu.badge_id
            """, [tuple(self.ids), tuple(self.env.companies.ids)])
```